### ISSUE TITLE
Add ability to configure certain errors to only be reported as warnings

### ISSUE THREAD

**Issue by RebeccaStevens** [webpro/knip#94]
Currently different types of issues can either be reported as errors or excluded.

It would be nice to be able have an in between level were issues are reported as warnings. Ideally this should all be configurable in the config file.

Any unlisted issue types can default to "error".

Example Config:

```jsonc
"issues": {
	"dependencies": "error",
	"enumMembers": "off", // The same as using `--exclude enumMembers`
	"exports": "warn",
	"files": "error",
	"types": "warn",
	"unlisted": "error"
}
```

A warning will print a warning message with `console.warn` but won't cause a non-zero exit code.

**Comment by webpro** [comment]
I think this is a very good idea. Atm I like to focus a bit more on getting v2 out, and then this can be a nice non-breaking feature soon after.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.6.0. See [Release 2.6.0](https://github.com/webpro/knip/releases/tag/2.6.0) for release notes._
